src/Mailgun/Api/HttpApi.php

```diff
@@ -56,38 +56,58 @@ public function __construct(HttpClient $httpClient, RequestBuilder $requestBuild
     }
 
     /**
-     * Attempts to safely deserialize the response into the given class.
-     * If the HTTP return code != 200, deserializes into SimpleResponse::class
-     * to contain the error message and any other information provided.
-     *
      * @param ResponseInterface $response
-     * @param string            $className
+     * @param string            $class
      *
-     * @throws HttpClientException
+     * @return mixed|array|ResponseInterface
```

Comment:
Are you sure we need to change the `@return` adding `mixed`? That looks wrong to me, if we don't know the type of content we're returning